sos/report/plugins/system.py

```diff
@@ -47,6 +48,42 @@ def setup(self):
                                 "environment_varnames",
                                 plug_dir=True)
 
+        sys_proxy_content = self._check_system_proxy_setting()
+
+        self.add_cmd_output(f'echo "{sys_proxy_content}"',
+                            suggest_filename="system_proxy_setting",
+                            root_symlink="system-proxy-setting",
+                            tags="sys_proxy")
+
+    def _check_system_proxy_setting(self):
+        sys_proxy_env_vars = [
+            'http_proxy',
+            'https_proxy',
+            'HTTP_PROXY',
+            'HTTPS_PROXY',
+        ]
+        sys_proxy_settings = {
+            var: os.environ.get(var)
+            for var in sys_proxy_env_vars
+            if os.environ.get(var)
+        }
+
+        def scrub_username_password(url):
+            """
+            Replace username:password@ with ********@ in proxy URL if present
+            """
+            return re.sub(r'//([^/@]+)@', '//********@', url)
```

Comment:
We have obfuscation of proxy URL credentials on several places of code already. Isn't it worth unifying it via one auxiliary method, @TurboTurtle ?